packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -1485,15 +1553,21 @@ export class GarbageCollector implements IGarbageCollector {
 }
 
 /**
- * Gets the garbage collection state from the given snapshot tree. The GC state may be written into multiple blobs.
- * Merge the GC state from all such blobs and return the merged GC state.
+ * Gets the garbage collection data from the given snapshot tree. It contains GC state and tombstone state.
+ * The GC state may be written into multiple blobs. Merge the GC state from all such blobs into one.
  */
-async function getGCStateFromSnapshot(
+async function getGCDataFromSnapshot(
     gcSnapshotTree: ISnapshotTree,
     readAndParseBlob: ReadAndParseBlob,
-): Promise<IGarbageCollectionState> {
+) {
     let rootGCState: IGarbageCollectionState = { gcNodes: {} };
+    let tombstones: string[] | undefined;
     for (const key of Object.keys(gcSnapshotTree.blobs)) {
+        if (key === gcTombstoneBlobKey) {
```

Comment:
This change needs to be saturated as N-1 right? Or will N-1 clients of today be able to read the tombstones logic?

Reply:
It should not break anything. Older client won't read tombstone data even if it exists. They will only read blob staring with `gcBlobPrefix` and ignore tombstone blobs.